### DISCUSSION TITLE
strongSwan upgrade to new version (build from source code)

### DISCUSSION THREAD

**Discussion by JimChen111** [strongswan/strongswan#2488] [Q&A] [answered]
Hi,

I have a question regarding doing strongSwan upgrade. If I install strongSwan from the source code with `./configure`, is the only option that I have removing the entire package first and then reinstall the new version from scratch? If that's the case, the old strongSwan configs cannot be retained, either? So I have to either recreate the strongSwan configs for policies or I need to manually save/copy the old configs to somewhere else and then move them back after the new strongSwan is installed. Is this correct?

Thanks in advance!

**Comment by tobiasbrunner** [comment, marked as answer]
> If I install strongSwan from the source code with `./configure`, is the only option that I have removing the entire package first and then reinstall the new version from scratch?

Wat exactly do you refer to with "removing the entire package"? Anyway, if you use the same configure options (in particular if you don't remove any plugins), you can just install over the existing installation with `make install`.

> If that's the case, the old strongSwan configs cannot be retained, either?

`make install` does not overwrite any existing config files (nor does `make uninstall` remove any in case you decide to run it).

**Reply by JimChen111** [reply]
Oh that's wonderful. So basically, if we don't need to modify the configure params (for dirs, plugins, etc), we don't even need to uninstall anything, and just `make install` the new strongSwan version to override the existing one? That'll be quite convenient.

**Reply by tobiasbrunner** [reply]
Yes, that should generally work fine.

**Reply by JimChen111** [reply]
Thank you for the great answer, Tobias!
